src/main/java/com/influxdb/v3/client/internal/VectorSchemaRootConverter.java

```diff
@@ -95,45 +97,141 @@ PointValues toPointValues(final int rowNumber,
+                } else {
+                    p.setNullField(fieldName);
+                    LOG.warning(String.format("Value of %s is not an Integer", fieldName));
+                }
+                break;
+            case "iox::column_type::field::float":
+                if (value instanceof Double) {
+                    p.setFloatField(fieldName, TypeCasting.toDoubleValue(value));
+                } else {
+                    p.setNullField(fieldName);
+                    LOG.warning(String.format("Value of %s is not a Double", fieldName));
+                }
+                break;
+            case "iox::column_type::field::string":
+                if (value instanceof String || value instanceof Text) {
+                    p.setStringField(fieldName, TypeCasting.toStringValue(value));
+                } else {
+                    p.setNullField(fieldName);
+                    LOG.warning(String.format("Value of %s is not a String", fieldName));
+                }
+                break;
+            case "iox::column_type::field::boolean":
+                if (value instanceof Boolean) {
+                    p.setBooleanField(fieldName, (Boolean) value);
+                } else {
+                    p.setNullField(fieldName);
+                    LOG.warning(String.format("Value of %s is not a Boolean", fieldName));
+                }
+                break;
+            default:
+                p.setField(fieldName, value);
+                break;
+        }
+    }
+
+    /**
+     * Get array of values from VectorSchemaRoot.
+     *
+     * @param vector    The data return from InfluxDB.
+     * @param rowNumber The row number of data
+     * @return  An array of Objects represent for a row of data
+     */
+    public Object[] getArrayObjectFromVectorSchemaRoot(final VectorSchemaRoot vector, final int rowNumber) {
+        List<Object> row = new ArrayList<>();
+        for (FieldVector fieldVector : vector.getFieldVectors()) {
+            var field = fieldVector.getField();
+            var metaType = field.getMetadata().get("iox::column::type");
+            String valueType = metaType != null ? metaType.split("::")[2] : null;
+            String fieldName = field.getName();
+
+            Object value = fieldVector.getObject(rowNumber);
+            if (value == null) {
+                row.add(null);
+                continue;
+            }
+
+            if ("field".equals(valueType)) {
+                switch (metaType) {
+                    case "iox::column_type::field::integer":
+                    case "iox::column_type::field::uinteger":
+                        if (value instanceof Long) {
+                            row.add(TypeCasting.toLongValue(value));
+                        } else {
+                            row.add(null);
+                            LOG.warning(String.format("Value of %s is not an Integer", fieldName));
```

Comment:
Set origin value instead of `null`. Add LOG statement to first place:

```suggestion
                            LOG.warning(String.format("Value of %s is not an Integer", fieldName));
                            row.add(null);
```

Apply to all variants.